s2/library/js/s2geometry.js

```diff
@@ -0,0 +1,1924 @@
+/// S2 Geometry functions
```

Comment:
This looks like a modified version from https://git.coolaj86.com/coolaj86/s2-geometry.js/src/branch/master/src/s2geometry.js and I'm not too happy with the approximation (it's neither well supported upstream nor here). How hard it would be to cover the basic use cases with the official C++ library (https://github.com/google/s2geometry)?